### DISCUSSION TITLE
Fully Dynamic Routing

### DISCUSSION THREAD

**Discussion by e-karma** [vercel/next.js#11931] [Help]
**Objective:** to achieve the NextJS equivalent to Gatsby's gatsby-node.js, 'createPage' function.  

I'm hoping to switch from Gatsby, but running into a rather important blocker. Here's how the code works in my existing Gatsby App.  

1. Loop over an array of data driven 'pages' from the CMS and call the 'createPage' function on each page.
2. Assign a page 'template' (react component) for each page based on data and business logic inside this loop

I have the above working in a bit more complicated way that aligns with your documentation, but the home page (when the url is an empty string '', implying the home route) does not work unless I add a 'index.js' file in the page directory.  

This makes the most important page static, which doesn't work for my use-case. 

Has anyone experienced this and solved for it?  If a code example exists that could be shared, where the INDEX (url.com/) route is generated dynamically, that would be very helpful.   

My approach has the reuseable logic in the pages/[...url].js file that includes both getStaticPaths & getStaticProps functions based on documentation.

**Comment by e-karma** [comment]
@timneutkens any insight would be appreciated!

**Reply by CaptainT33mo** [reply]
@e-karma check out this question that I asked https://github.com/zeit/next.js/discussions/11936.
I am yet to implement this in my project. Let us know after implementing if that solution works.

**Comment by miloofcroton** [comment]
No offense, but this seems like an anti-pattern to me. Is there a reason you don't want to add a resource identifier to your url (e.g. /movies/batman/actors instead of just /batman/actors if I am understanding you correctly)?
